pydicom_seg/reader.py

```diff
@@ -167,8 +167,14 @@ def read(self, dataset: pydicom.Dataset) -> SegmentReadResult:
 
             # Iterate over all frames and check for referenced segment number
             for frame_idx, pffg in enumerate(dataset.PerFrameFunctionalGroupsSequence):
-                if segment_number != pffg.SegmentIdentificationSequence[0].ReferencedSegmentNumber:
-                    continue
+                if hasattr(pffg, 'SegmentIdentificationSequence'):
+                    sis = pffg.SegmentIdentificationSequence[0]
+                elif hasattr(dataset.SharedFunctionalGroupsSequence[0], 'SegmentIdentificationSequence'):
+                    sis = dataset.SharedFunctionalGroupsSequence[0].SegmentIdentificationSequence[0]
```

Comment:
I would move this part up, out of the loop, since it is constant and should only be extracted once. For instance 
```python
shared_sis = dataset.SharedFunctionalGroupsSequence[0].get('SegmentIdentificationSequence')
```